core/src/autogluon/core/utils/feature_selection.py

```diff
@@ -0,0 +1,497 @@
+            logger.log(30, f"\tExpected model fit time: {round(self.model_fit_time, 4)}s, and expected candidate generation time: {round(time_budget_fi, 2)}s.")
+            logger.log(30, f"\tFit 1 ({round(self.model_fit_time, 4)}s): Current score is {best_info['score']}.")
+            if self.time_limit < self.model_fit_time + time_budget_fi:
+                logger.log(30, f"\tNo time to perform the next pruning round (remaining: {self.time_limit}, needed: {self.model_fit_time + time_budget_fi}).")
+                self._debug_info['exceptions'].append(f"exiting after 1 model fit (remaining: {self.time_limit}, needed: {self.model_fit_time+time_budget_fi})")
+                raise TimeLimitExceeded
+
+            importance_df = None
+            while True:
+                index = index + 1
+                model_name = f"{self.base_model.name}_{index}"
+                prev_candidate_features = candidate_features
+                prioritize_fi = set([feature for feature in best_info['features'] if self.noise_prefix in feature])
+                fn_args = {'X': X_fi, 'y': y_fi, 'model': best_info['model'], 'time_budget': time_budget_fi, 'features': best_info['features'],
+                           'n_subsample': n_fi_subsample, 'prune_threshold': prune_threshold, 'prune_ratio': prune_ratio, 'prioritized': prioritize_fi}
+                fn_args.update(self.get_extra_fn_args(**kwargs))
+                candidate_features, importance_df, success, prune_time = self.compute_next_candidate(fn_args, time_budget_fi, prev_candidate_features,
+                                                                                                     importance_df)
+                if not success:
+                    logger.log(30, f"\tTime is up while computing feature importance or there are no more features to prune. Ending...")
+                    break
+                curr_model, score, fit_time = self.fit_score_model(X, y, X_val, y_val, candidate_features, model_name, **kwargs)
+
+                if score >= best_info['score'] * (1 + improvement_threshold):
+                    logger.log(30, f"\tFit {index} ({fit_time}s): Current score {score} is better than best score {best_info['score']}. Updating model.")
+                    logger.log(30, f"\tOld # Features: {len(best_info['features'])} / New # Features: {len(candidate_features)}.")
+                    prev_best_model = best_info['model']
+                    best_info = {'model': curr_model, 'features': candidate_features, 'score': score, 'index': index}
+                    if not self.keep_models and prev_best_model.name != self.original_model.name:
+                        prev_best_model.delete_from_disk()
+                    self._debug_info['index_trajectory'].append(True)
+                else:
+                    logger.log(30, f"\tFit {index} ({fit_time}s): Current score {score} is not better than best score {best_info['score']}. Retrying.")
```

Comment:
"Fit k" is not very informative output for user, consider more clear print statement, eg. "Fit in kth round of feature pruning".  

For all print statements, I'd assume the user does not know what is happening internally, so make them as informative as possible while still concise.